src/librustc_codegen_ssa/mir/mod.rs

```diff
@@ -423,7 +427,20 @@ fn arg_local_refs<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>>(
             bx.store_fn_arg(arg, &mut llarg_idx, tmp);
             LocalRef::Place(tmp)
         }
-    }).collect()
+    }).collect::<Vec<_>>();
+
+    if fx.instance.def.requires_caller_location(bx.tcx()) {
+        assert_eq!(
+            fx.fn_abi.args.len(), args.len() + 1,
+            "#[track_caller] fn's must have 1 more argument in their ABI than in their MIR",
+        );
+        let arg = &fx.fn_abi.args.last().unwrap();
+        let place = PlaceRef::alloca(bx, arg.layout);
+        bx.store_fn_arg(arg, &mut llarg_idx, place);
+        fx.caller_location = Some(place);
```

Comment:
You can either `load_operand` or just require that the arg is passed directly, as an immediate, and create an `OperandRef` from it.
This would avoid a load on every use of it.

Reply:
Cool! Could you give me a little more detail about how to do so?

Reply:
Like this (non-direct is impossible): https://github.com/rust-lang/rust/blob/052d0edbc1c2572ba735ef911a5ab7e9c2a703bc/src/librustc_codegen_ssa/mir/mod.rs#L383-L388